addons/mail/models/mail_channel.py

```diff
@@ -90,6 +105,113 @@ def default_get(self, fields):
              "Use this field anywhere a small image is required.")
     is_subscribed = fields.Boolean(
         'Is Subscribed', compute='_compute_is_subscribed')
+    moderation = fields.Boolean(string='Moderate this channel')
+    moderator_ids = fields.Many2many('res.users', 'mail_channel_moderator_rel', string='Moderators')
+    moderated_email_ids = fields.One2many('channel.moderated.emails', 'channel_id', string='Moderated Emails')
+    auto_notification = fields.Boolean(string="Automatic notification", help="People receive an automatic notification about their message being waiting for moderation.")
+    notification_message = fields.Text(string="Notification message")
+    send_guidelines = fields.Boolean(string="Send guidelines to new subscribers", help="Newcomers on this moderated channel will automatically receive the guidelines.", default=False)
+    guidelines = fields.Text(string="Guidelines")
+
+    @api.constrains('moderator_ids')
+    def _check_moderators_have_email_addresses(self):
```

Comment:
Naming: just to simplify code, I think we could rename it to something like _check_moderator_email